apps/swirl-docs/src/components/Icons/IconDownload.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  autoUpdate,
+  flip,
+  FloatingFocusManager,
+  shift,
+  useClick,
+  useDismiss,
+  useFloating,
+  useInteractions,
+  useRole,
+} from "@floating-ui/react";
+import {
+  SwirlActionList,
+  SwirlActionListItem,
+  SwirlIconChevronRight,
+  SwirlSpinner,
+} from "@getflip/swirl-components-react";
+import classNames from "classnames";
+import { FunctionComponent, useState } from "react";
+import { IconData } from "src/pages/icons";
+import { initializeIconDownload } from "./utils";
+
+interface IconDownloadProps {
+  icon: IconData;
+}
+
+export const IconDownload: FunctionComponent<IconDownloadProps> = ({
+  icon,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [iconPixelSize, setIconPixelSize] = useState(24);
+  const [isLoadingIconDownload, setIsLoadingIconDownload] = useState(false);
+
+  const { reference, floating, strategy, context } = useFloating({
+    open: isOpen,
+    onOpenChange: setIsOpen,
+    middleware: [flip({ fallbackAxisSideDirection: "end" }), shift()],
+    whileElementsMounted: autoUpdate,
+  });
+
+  const click = useClick(context);
+  const dismiss = useDismiss(context);
+  const role = useRole(context);
+
+  const { getReferenceProps, getFloatingProps } = useInteractions([
+    click,
+    dismiss,
+    role,
+  ]);
+
+  function handleActionItemClick(size: number): void {
+    setIconPixelSize(size);
+    setIsOpen(false);
```

Comment:
**thought:** Should we start the download here immediately?

Reply:
good idea. implemented